src/Microsoft.AspNetCore.SignalR/HubEndPoint.cs

```diff
@@ -232,7 +232,7 @@ private void DiscoverHubMethods()
         private static bool IsHubMethod(MethodInfo m)
         {
             // TODO: Add more checks
-            return m.IsPublic;
+            return m.IsPublic && typeof(Hub<IClientProxy>) != m.GetBaseDefinition().DeclaringType;
```

Comment:
We care about this so clients can't call methods like `OnConnectedAsync` and `OnDisconnectedAsync`